File: src/test/console/php/facebook-webdriver/AndroidWebTest.php
<?php
require_once('vendor/autoload.php');
use Facebook\WebDriver\Remote\RemoteWebDriver;
use Facebook\WebDriver\WebDriverBy;
include("utils/AutomationUtils.php");
include("services/DeviceService.php");

class AndroidWebTest extends PHPUnit_Framework_TestCase {
  protected $driver;
  public $wrong_username_msg = 'Your username is invalid!';
  public $success_msg = 'You logged into a secure area!';

  public function setUp() {
    include("configs.php");

    $device = getOnlineDevice('Android');
    $caps = desiredCapabilitiesAndroidWeb($device);

    $this->driver = RemoteWebDriver::create(
      kobitonServerUrl(),
      $caps,
      $connection_timeout_in_ms = 1200000,
      $request_timeout_in_ms = 1200000
    );
    $this->driver->manage()->timeouts()->implicitlyWait(60); // seconds
  }
   /*
    Steps:
    1. Login invalid user
    2. Login successfully with correct username and passoword
    */

  public function test_android_web() {
    $this->verify_login_invalid_username();
    $this->verify_login_successfully();
  }

  public function verify_login_invalid_username() {
    print 'should return error when we input wrong username';
    $this->login('foo', 'SuperSecretPassword!');
    $this->assertContains($this->wrong_username_msg, $this->getMessage());
  }

  public function verify_login_successfully() {
    print 'should run test successfully with correct username and password';
    $this->login('tomsmith', 'SuperSecretPassword!');
    $this->assertContains($this->success_msg, $this->getMessage());
  }

  function login($username, $password) {
    $this->driver->get("https://the-internet.herokuapp.com/login");
    $this->driver->findElement(WebDriverBy::id('username'))->sendKeys($username);
    $this->driver->findElement(WebDriverBy::id('password'))->sendKeys($password);
    $this->driver->findElement(WebDriverBy::xpath("//form[@name='login']"))->submit();
  }

  public function getMessage() {
    return $this->driver->findElement(WebDriverBy::id('flash'))->getText();
  }

  public function tearDown() {
    if ($this->driver) {
      $this->driver->quit();
    }
    parent::tearDown();
  }
}

File: ruby_rspec/androidWebTest.rb
require "selenium-webdriver"
require_relative "configs" 
require_relative "retry"

describe "Android Web Test" do

  before(:all) do 
    caps = DesiredCapabilities.new
    @driver = Selenium::WebDriver.for(:remote, :url => caps.desired_caps_android_web[:url],
    :desired_capabilities => caps.desired_caps_android_web[:caps])
    @wait = Selenium::WebDriver::Wait.new(:timeout => 10)
    @driver.manage.timeouts.implicit_wait = 300 
  end

  after(:all) do
    @driver.quit
  end

  it "wrong username" do
    wrong_username_msg = 'Your username is invalid!'
    puts 'Test case 1: it should return error when we input wrong username'
    login('foo', 'SuperSecretPassword!')
    expect(get_message).to include(wrong_username_msg)
  end

  it "wrong password" do
    wrong_password_msg = 'Your password is invalid!'
    puts 'Test case 2: it should return error when we input wrong password'
    login('tomsmith', 'SuperSecretPassword')
    expect(get_message).to include(wrong_password_msg)
  end

  it "login successfully" do
    success_msg = 'You logged into a secure area!'
    puts 'Test case 3: it should run test successfully with correct username and password'
    login('tomsmith', 'SuperSecretPassword!')
    expect(get_message).to include(success_msg) 
  end
end

def login(username, password)
  @driver.navigate.to('http://the-internet.herokuapp.com/login')
  @driver.find_element(:id, 'username').send_keys(username)
  @driver.find_element(:id, 'password').send_keys(password)
  @driver.find_element(:xpath, "//form[@name='login']").submit
end

def get_message
  return @driver.find_element(:id, 'flash').text
end

